looker/resource_looker_user_attribute.go

```diff
@@ -0,0 +1,241 @@
+	}
+
+	if userAttributes.Id == nil {
+		return diag.Errorf("user attribute has missing id")
+	}
+	d.SetId(*userAttributes.Id)
+
+	return resourceUserAttributeRead(ctx, d, c)
+}
+
+func resourceUserAttributeRead(ctx context.Context, d *schema.ResourceData, c interface{}) diag.Diagnostics {
+	api := c.(*sdk.LookerSDK)
+
+	userAttributes, err := api.UserAttribute(d.Id(), "", nil)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	var defaultValue *string
+	// if hidden, default value cannot be retrieved from API
+	if *userAttributes.ValueIsHidden {
+		// use has change since API cannot give us default value
+		if d.HasChange("default_value") {
+			defaultValue = conv.PString("DEFAULT_IS_SET")
+		} else {
+			defaultValue = conv.PString(d.Get("default_value").(string))
+		}
+	} else {
+		defaultValue = userAttributes.DefaultValue
+	}
+
+	var userAccess string
+	if *userAttributes.UserCanView {
+		userAccess = "View"
+	} else if *userAttributes.UserCanEdit {
+		userAccess = "Edit"
+	} else {
+		userAccess = "None"
+	}
+
+	var domainsWhitelistSlice []string
+	if userAttributes.HiddenValueDomainWhitelist != nil {
+		domainsWhitelistSlice = strings.Split(*userAttributes.HiddenValueDomainWhitelist, ",")
+	}
+
+	result := multierror.Append(
+		d.Set("id", userAttributes.Id),
+		d.Set("name", userAttributes.Name),
+		d.Set("label", userAttributes.Label),
+		d.Set("data_type", userAttributes.Type),
+		d.Set("hidden", userAttributes.ValueIsHidden),
+		d.Set("default_value", defaultValue),
+		d.Set("user_access", conv.PString(userAccess)),
+		d.Set("domain_whitelist", conv.PSlices(domainsWhitelistSlice)),
+	)
+
+	return diag.FromErr(result.ErrorOrNil())
+}
+
+func resourceUserAttributeUpdate(ctx context.Context, d *schema.ResourceData, c interface{}) diag.Diagnostics {
+	api := c.(*sdk.LookerSDK)
+
+	userAttrs, err := buildUserAttributeInput(d)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	_, err = api.UpdateUserAttribute(d.Id(), *userAttrs, "id", nil)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	return resourceUserAttributeRead(ctx, d, c)
+}
+
+func resourceUserAttributeDelete(ctx context.Context, d *schema.ResourceData, c interface{}) diag.Diagnostics {
+	api := c.(*sdk.LookerSDK)
+
+	_, err := api.DeleteUserAttribute(d.Id(), nil)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	return nil
+}
+
+func buildUserAttributeInput(d *schema.ResourceData) (*sdk.WriteUserAttribute, error) {
+	userAttr := sdk.WriteUserAttribute{
+		Name:          *conv.PString(d.Get("name").(string)),
+		Label:         *conv.PString(d.Get("label").(string)),
+		Type:          *conv.PString(d.Get("data_type").(string)),
+		ValueIsHidden: conv.PBool(d.Get("hidden").(bool)),
+		DefaultValue:  conv.PString(d.Get("default_value").(string)),
+	}
+
+	user_access, ok := d.Get("user_access").(string)
+	if !ok {
+		return nil, errors.New("user_access is not a string")
+	}
+
+	switch user_access {
+	case "View":
+		userAttr.UserCanView = conv.PBool(true)
```

Comment:
In the case of changing from `Edit` to `View`, you may need to unset the `Edit` permissions back to false?